packages/react-server/src/ReactFlightServer.js

```diff
@@ -214,10 +216,12 @@ function attemptResolveElement(
     }
     // This might be a built-in React component. We'll let the client decide.
     // Any built-in works as long as its props are serializable.
+    // $FlowFixMe unsure how this return type is supposed to match ReactModel
     return [REACT_ELEMENT_TYPE, type, key, props];
   } else if (type != null && typeof type === 'object') {
     if (isModuleReference(type)) {
       // This is a reference to a client component.
+      // $FlowFixMe unsure how this return type is supposed to match ReactModel
       return [REACT_ELEMENT_TYPE, type, key, props];
```

Comment:
What branch of `ReactModel` is this supposed to match?

```
export type ReactModel =
  | React$Element<any>
  | string
  | boolean
  | number
  | null
  | Iterable<ReactModel>
  | ReactModelObject;
```

Is this an `Iterable<ReactModel>` and the recursive `ReactModel` is missing `symbol` for the `REACT_ELEMENT_TYPE`?

Reply:
Seems that was the right fix.